modules/components/src/QuickSearch/QuickSearch.js

```diff
@@ -39,39 +39,45 @@ const toggleValue = ({ sqon, setSQON, primaryKeyField, primaryKey }) =>
 	);
 
 const inputRef = React.createRef();
-const QuickSearch = ({
-	apiFetcher,
-	className,
-	documentType,
-	DropdownItemComponent = DropdownItem,
-	headerTitle = 'Search by ID',
-	Icon = FaSearch,
-	InputComponent = TextInput,
-	LoadingIcon = FaSearch,
-	PinnedValueComponent = SQONBubble,
-	placeholder = 'Quick Search',
-	searchLowercase = false,
-	searchTextDelimiters = ['\\s', ','],
-	setSQON,
-	sqon,
-	style = emptyObj,
-	...props
-}) => {
+const QuickSearch = ({ ...props }) => {
 	const [value, setValue] = useState('');
+	const { apiFetcher, documentType, sqon, setSQON } = useDataContext();
 	const {
 		components: {
 			QuickSearch: {
-				FilterInput: themeQuickSearchFilterInput = emptyObj,
+				DropDownItems: { DropdownItemComponent = DropdownItem } = emptyObj,
+				FilterInput: {
+					css: themeFilterInputCSS,
+					Icon = FaSearch,
+					InputComponent = TextInput,
+					LoadingIcon = FaSearch,
+					placeholder: themeFilterInputPlaceholder = 'Quick Search',
+				} = emptyObj,
+				QuickSearchQuery: {
+					searchLowercase: themeQuickSearchQuerySearchLowercase = false,
+					searchTextDelimiters: themeQuickSearchQuerySearchTextDelimiters = ['\\s', ','],
+				} = emptyObj,
+				QuickSearchWrapper: {
+					headerTitle: themeQuickSearchWrapperHeaderTitle = 'Search by ID',
+					className: themeQuickSearchWrapperClassName = '',
+					style: themeQUickSearchWrapperStyle = emptyObj,
+				} = emptyObj,
 				PinnedValues: {
 					enabled: themePinnedValuesEnabled = true,
-					theme: themePinnedValuesTheme,
+					PinnedValueComponent = SQONBubble,
+					css: themePinnedValuesCss,
+					...themePinnedValuesTheme
 				} = emptyObj,
 			} = emptyObj,
 		} = emptyObj,
 	} = useThemeContext({ callerName: 'QuickSearch' });
 
 	return (
-		<QuickSearchWrapper displayName={headerTitle} className={className} style={style}>
+		<QuickSearchWrapper
+			displayName={themeQuickSearchWrapperHeaderTitle}
+			className={themeQuickSearchWrapperClassName}
+			style={themeQUickSearchWrapperStyle}
+		>
 			<QuickSearchFieldsQuery
 				{...props}
```

Comment:
following up with the previous comment, if anyone passes `theme` into the parent component, it'll be passed down here. It may be a good idea for now to just remove that `props` spread altogether, and we can rebuild this thing as needed later.